lib/exercises.rb

```diff
@@ -1,19 +1,54 @@
 
 # This method will return an array of arrays.
 # Each subarray will have strings which are anagrams of each other
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: o(n^2)
+# Space Complexity: o(n) => created 1 new hash and 1 new array. It should be o(2n), but we can ignor the constant
 
 def grouped_anagrams(strings)
-  raise NotImplementedError, "Method hasn't been implemented yet!"
+  anagrams_hash = {}
+
+  strings.each do |string|
+    sorted_s = string.chars.sort.join
+    if !anagrams_hash.has_key? (sorted_s)
```

Comment:
Can be simplified a bit
```suggestion
    if !anagrams_hash.[sorted_s].nil?
```